### PR TITLE
added completions for sad and added note in changelog

### PR DESCRIPTION
**Pull request by bagohart**
## Description
I added tab completion for [sad](https://github.com/ms-jpq/sad).

Fixes issue ms-jpq/sad#153

## TODOs:

- [ ] Changes to fish usage are reflected in user documentation/manpages.
- [ ] Tests have been added for regressions fixed
- [x] User-visible changes noted in CHANGELOG.rst

### CODE REVIEW

File: share/completions/sad.fish [hunk 1]
```diff
@@ -0,0 +1,9 @@
+complete sad    -s 0 -l read0                    -d 'Use \0 as stdin delimiter'
+complete sad    -s e -l exact                    -d 'String literal mode'
+complete sad -x -s f -l flags                    -d "Regex flags: use `--help` instead of `-h` to see details"
```
Comment (faho) [inline share/completions/sad.fish:3]: What does this mean?

Does this treat `sad -h` different from `sad --help`? Is the `-h` version short help?

That's... unorthodox.

File: share/completions/sad.fish [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+complete sad -x      -l fzf     -a "never"       -d "Additional Fzf options, disable = never"
```
Comment (faho) [inline share/completions/sad.fish:4]: What does this disable, exactly?

File: share/completions/sad.fish [hunk 3]
```diff
@@ -0,0 +1,9 @@
+complete sad    -s 0 -l read0                    -d 'Use \0 as stdin delimiter'
```
Comment (faho) [inline share/completions/sad.fish:1]: This formatting will be completely obliterated the next time we `fish_indent` this file.

It's not worth it to keep alignment.

File: share/completions/sad.fish (hunk 1 above)
Comment (bagohart) [reply share/completions/sad.fish]: Yes, exactly.

File: share/completions/sad.fish (hunk 2 above)
Comment (bagohart) [reply share/completions/sad.fish]: including `--fzf=never` disables fzf and previews the changes with `$GIT_PAGER`.

File: share/completions/sad.fish (hunk 3 above)
Comment (bagohart) [reply share/completions/sad.fish]: okay, I ran `fish_indent` on it.

File: share/completions/sad.fish (hunk 2 above)
Comment (faho) [reply share/completions/sad.fish]: And passing other values gives options to fzf?

File: share/completions/sad.fish (hunk 1 above)
Comment (faho) [reply share/completions/sad.fish]: Okay, in that case `-h` and `--help` are separate options, and they should then be separate to complete.

`-h` is "short help" and `--help` is "long help" or similar.

File: share/completions/sad.fish (hunk 2 above)
Comment (bagohart) [reply share/completions/sad.fish]: Yes, e.g. try this in `fish-shell/share/completions/`: `ls sad.fish | sad --fzf="--header 'lol?'" "complete" "nope"` overwrites the fzf header.

File: share/completions/sad.fish (hunk 1 above)
Comment (bagohart) [reply share/completions/sad.fish]: I guess that makes sense, done.

**Comment by faho** [comment]
Merged, thanks!

**Pull request merged**